regtests/client/python/cli/command/catalog_roles.py

```diff
@@ -46,6 +46,8 @@ class CatalogRolesCommand(Command):
     catalog_role_name: str
     principal_role_name: str
     properties: Optional[Dict[str, StrictStr]]
```

Comment:
What does `properties` do now? Can we remove it?

Reply:
i.e. could you use `set-property` during `create` as well?

Reply:
My thinking for preserving `--property` for `create` is to emphasize the contrasting "declarative" semantics of CREATE vs the choice of having an "imperative" semantic for UPDATE. Looking at examples of other CLIs with a similar split, the same pattern is used:

    gcloud compute instances create ... --labels=...
    gcloud compute instances update ... --update-labels=... --remove-labels=...

And similarly `--update-labels` and `--remove-labels` aren't available in the `create` command, while `--labels` is not available in the `update` command.

Reply:
That makes sense. My initial hesitation was just having two properties that do the same thing. 

Lets keep this, but I think we should reject the "invalid" inputs then:
`catalogs update some_catalog --property k=v` 
`catalogs update some_catalog --set-property k=v`

Reply:
Agreed, this test case in test_cli_parsing.py covers the rejection of `--property`:

https://github.com/apache/polaris/pull/404/files#diff-437c53a66ce1039f3ff70bef07bb0325924edf2b8f2c864762969e6ecdde3ecfR58

        with self.assertRaises(SystemExit) as cm:
            Parser.parse(['catalogs', 'update', 'catalog_name', '--property', 'foo=bar'])
        self.assertEqual(cm.exception.code, INVALID_ARGS)